<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="bump">

	<link name="bump">
     <inertial>
      <origin xyz="0 0 0.5" rpy="0 0 0" /> 
      <mass value="10.0" />
      <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />
    </inertial>
    <visual>
      <origin xyz="0 0 0.5" rpy="0 0 0" /> 
      <geometry>
        <box size="0.03 0.1 0.1" />
      </geometry>
    </visual>
  </link>

	<joint name="bump" type="fixed">
		<origin xyz="3.9 0 0" rpy="0 0 0"/>
		<parent link="base_footprint"/>
		<child link="bump"/>
	</joint>
		
		
	<gazebo reference="bump">
    <sensor name="bump" type="contact">
			<always_on>1</always_on>
    	<update_rate>100</update_rate>
			<contact>
		  	  <collision>base_footprint_fixed_joint_lump__chassis_collision_collision</collision>
			</contact>
		  <plugin filename="libgazebo_ros_bumper.so" name="bump_gazebo_ros_bumper_controller">
		    <frameName>/vehicle/bump</frameName>
		    <bumperTopicName>bumper_state</bumperTopicName>
		  </plugin>
  	</sensor>
  </gazebo>

</robot>
